akka-http-core/src/main/scala/akka/http/impl/settings/ConnectionPoolSettingsImpl.scala

```diff
@@ -26,7 +28,8 @@ private[akka] final case class ConnectionPoolSettingsImpl(
   idleTimeout:                       Duration,
   connectionSettings:                ClientConnectionSettings,
   poolImplementation:                PoolImplementation,
-  responseEntitySubscriptionTimeout: Duration)
+  responseEntitySubscriptionTimeout: Duration,
+  hostOverrides:                     immutable.Seq[(Regex, ConnectionPoolSettings)])
```

Comment:
Do we allow nested `hostOverrides`? If no, we should add a require clause here to check that. Something like

```scala
require(hostOverrides.isEmpty || hostOverrides.forall(_._2.isEmpty))
```